### PR TITLE
apply ordering to RBAC resources in FeatureGates package

### PR DESCRIPTION
**Pull request by yharish991**
Signed-off-by: Harish Yayi <yharish991@gmail.com>

### What this PR does / why we need it
This PR adds kapp builtin rules(https://carvel.dev/kapp/docs/latest/apply-ordering/) to FeatureGates rbac resources to make sure they are created in a proper order
ServiceAccount(tanzu-featuregates-manager-sa) and ClusterRole(tanzu-featuregates-manager-clusterrole) -> ClusterRoleBinding(tanzu-featuregates-manager-clusterrolebinding)

### Which issue(s) this PR fixes

Fixes https://github.com/vmware-tanzu/tanzu-framework/issues/1079

### Describe testing done for PR

1. Install kapp-controller [v0.28.0](https://github.com/vmware-tanzu/carvel-kapp-controller/releases/tag/v0.28.0)
2. Create PackageRepository CR
```
apiVersion: packaging.carvel.dev/v1alpha1
kind: PackageRepository
metadata:
  name: management.tanzu.vmware.com
  namespace: tkg-system
spec:
  fetch:
    imgpkgBundle:
      image: quay.io/hyayiv/management@sha256:76a447d44fdb6b31d8a40daf4d2d171441c2771229cf077fba0f094f8430e238
```
3. Create ServiceAccount for PackageInstall CR
```
apiVersion: v1
kind: ServiceAccount
metadata:
  name: tanzu-featuregates-package-sa
  namespace: tkg-system
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: tanzu-featuregates-package-cluster-role
rules:
  - apiGroups:
      - ""
    resources:
      - configmaps
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - ""
    resources:
      - serviceaccounts
      - services
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - apps
    resources:
      - deployments
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - apiextensions.k8s.io
    resources:
      - customresourcedefinitions
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - cert-manager.io
    resources:
      - issuers
      - certificates
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - admissionregistration.k8s.io
    resources:
      - validatingwebhookconfigurations
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - rbac.authorization.k8s.io
    resources:
      - clusterroles
      - clusterrolebindings
    verbs:
      - create
      - update
      - get
  - apiGroups:
      - config.tanzu.vmware.com
    resources:
      - featuregates
    verbs:
      - create
      - delete
      - get
      - list
      - patch
      - update
      - watch
  - apiGroups:
      - config.tanzu.vmware.com
    resources:
      - featuregates/status
    verbs:
      - get
      - patch
      - update
  - apiGroups:
      - config.tanzu.vmware.com
    resources:
      - features
    verbs:
      - get
      - list
      - watch
  - apiGroups:
      - config.tanzu.vmware.com
    resources:
      - features/status
    verbs:
      - get
      - patch
      - update
  - apiGroups:
      - ""
    resources:
      - namespaces
    verbs:
      - get
      - list
      - watch
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: tanzu-featuregates-package-cluster-rolebinding
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: tanzu-featuregates-package-cluster-role
subjects:
  - kind: ServiceAccount
    name: tanzu-featuregates-package-sa
    namespace: tkg-system
```
4. Create PackageInstall CR
```
apiVersion: packaging.carvel.dev/v1alpha1
kind: PackageInstall
metadata:
  name: tanzu-featuregates
  namespace: tkg-system
spec:
  packageRef:
    refName: featuregates.tanzu.vmware.com
    versionSelection:
      prereleases: {}
  serviceAccountName: tanzu-featuregates-package-sa
```
5. Check PackageInstall CR status
```
$ kubectl get packageinstalls.packaging.carvel.dev tanzu-featuregates -n tkg-system -o jsonpath="Status: {.status}"
```

### Release note

```release-note
Fixed FeatureGates package installation by applying ordering rules to FeatureGates RBAC resources
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Comment by rajathagasthya** [comment]
> What this PR does / why we need it
> 
> This PR adds kapp builtin rules(https://carvel.dev/kapp/docs/latest/apply-ordering/) to FeatureGates rbac resources to make sure they are created in a proper order
ServiceAccount(tanzu-featuregates-manager-sa) and ClusterRole(tanzu-featuregates-manager-clusterrole) -> ClusterRoleBinding(tanzu-featuregates-manager-clusterrolebinding)

@yharish991 This PR description would also be a great commit message detail. If you make updates to the PR, please consider adding it to the commit message.

**Review by rajathagasthya — COMMENTED** [review]
LGTM, but I'm not too familiar with these annotations. Would like someone who knows about it to give an approval.

File: packages/management/featuregates/bundle/config/upstream/rbac.yaml [hunk 1]
```diff
@@ -3,12 +3,16 @@ kind: ServiceAccount
 metadata:
   labels:
     app: tanzu-featuregates-manager
+  annotations:
+    kapp.k14s.io/change-group: "featuregates.config.tanzu.vmware.com/serviceaccount"
```
Comment (cppforlife) [inline packages/management/featuregates/bundle/config/upstream/rbac.yaml:7]: dont think we need to add SA to a group.

Comment (yharish991) [reply packages/management/featuregates/bundle/config/upstream/rbac.yaml:7]: ClusterRoleBinding has a reference to this SA, so just to be safe I added this to the group as well even though I did not see an error with SA and also i think it doesn't harm anything, but if you are sure that removing this annotation on SA wouldn't cause the issue I can remove it(the problem is its hard to test this fix as i don't hit this issue always)

Comment (yharish991) [reply packages/management/featuregates/bundle/config/upstream/rbac.yaml:7]: let's leave this as-is for now and can revisit if needed later. Also, I think kapp-controller will be fixing this in its new version, so we can come revisit then if needed

**Review by jmoroski — APPROVED** [review]

**Pull request merged**